compiler/rustc_typeck/src/lib.rs

```diff
@@ -534,7 +534,7 @@ pub fn check_crate(tcx: TyCtxt<'_>) -> Result<(), ErrorReported> {
     check_unused::check_crate(tcx);
     check_for_entry_fn(tcx);
 
-    if tcx.sess.err_count() == 0 { Ok(()) } else { Err(ErrorReported) }
+    if tcx.sess.err_count() == 0 { Ok(()) } else { Err(ErrorReported::new_i_know_what_im_doing()) }
```

Comment:
`err_count` should give access to a `ErrorReported` for `> 0` counts (see other comments).

Reply:
I think we can just use `has_errors` here...